lib/internal/url.js

```diff
@@ -932,6 +935,60 @@ function urlToOptions(url) {
   return options;
 }
 
+function getPathFromURLWin32(url) {
+  var hostname = url.hostname;
+  var pathname = url.pathname;
+  if (hostname !== '') {
+    // If hostname is set, then we have a UNC path
```

Comment:
I'm not sure if we should doc the changes yet or not. I still haven't even been able to land the docs for the URL object itself yet.

And good point about the handling of pct-decoding in the hostname. We should likely be doing that here instead of relying on the OS to do it.